doc/Architecture.md

```diff
@@ -0,0 +1,53 @@
+# Architecture
+
+This section describes the overall architecture of Theia. 
+Theia is designed to work as a native desktop application as well as in the context of a browser and a remote server. To support both situations with a single source, Theia runs in two separate processes. Those processes are called _frontend_ and _backend_ respectively, and they communicate through JSON-RPC messages over WebSockets or REST APIs over HTTP. In the case of Electron, the backend, as well as the frontend, run locally, while in a remote context the backend would run on a remote host.
+
+Both the frontend and backend processes have their dependency injection container (see below) to which extensions can contribute.
+
+## Frontend
+
+The frontend process represents the client and renders the UI. In the browser, it simply runs in the rendering loop, while in Electron it runs in an Electron Window, which basically is a browser with additional Electron and node.js APIs. Therefore, frontend code may only access sources that depend on any API that can run in browsers.
+
+The startup of the frontend process will first load the modules of all contributing extensions before it obtains an instance of `FrontendApplication` from the resulting Dependency Injection (DI) container and call `start()` on it.
+
+## Backend
+
+ The backend process runs on Node.js. We use express as the HTTP server. The backend also serves the static code for the frontend. So the backend is started first and exposes a port to serve the frontend application to a requesting browser. For Electron we open an `Electron.BrowserWindow` which is a chromium with some additional Electron APIs.
+
```

Comment:
It's not clear to me what you mean here by 'serving the static code' like why is it static ? I assume it's not serving all of the frontend code since it needs some code to load the backend... I'm confused about what code it serves basically ?